instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/logging/LogbackAppenderApplicationListener.java

```diff
@@ -63,12 +63,25 @@ public void onApplicationEvent(ApplicationEvent event) {
       if (existingOpenTelemetryAppender.isPresent()) {
         reInitializeOpenTelemetryAppender(
             existingOpenTelemetryAppender, applicationEnvironmentPreparedEvent);
-      } else {
+      } else if (isLogbackAppenderAddable(applicationEnvironmentPreparedEvent)) {
         addOpenTelemetryAppender(applicationEnvironmentPreparedEvent);
       }
     }
   }
 
+  private static boolean isLogbackAppenderAddable(
+      ApplicationEnvironmentPreparedEvent applicationEnvironmentPreparedEvent) {
+    Boolean otelSdkDisableProperty =
+        evaluateBooleanProperty(applicationEnvironmentPreparedEvent, "otel.sdk.disabled");
```

Comment:
Perhaps it would make sense to add a variant of `evaluateBooleanProperty` that returns `boolean` and accepts the default value as an argument.

Reply:
For me it seems a little early to add this method today because (if I understand correctly):
* The method would be called at only two places
* The default value as an argument would be true in both cases